src/Core/System.php

```diff
@@ -292,6 +292,36 @@ public static function isDirectoryUsable($directory, $check_writable = true)
 		return true;
 	}
 
+	/**
+	 * Exit method used by asynchronous update modules
+	 *
+	 * @param string $o
+	 */
+	public static function htmlUpdateExit($o)
+	{
+		if (DI::pConfig()->get(local_user(), "system", "bandwidth_saver")) {
+			$replace = "<br />".DI::l10n()->t("[Embedded content - reload page to view]")."<br />";
+			$pattern = "/<\s*audio[^>]*>(.*?)<\s*\/\s*audio>/i";
+			$o = preg_replace($pattern, $replace, $o);
+			$pattern = "/<\s*video[^>]*>(.*?)<\s*\/\s*video>/i";
+			$o = preg_replace($pattern, $replace, $o);
+			$pattern = "/<\s*embed[^>]*>(.*?)<\s*\/\s*embed>/i";
+			$o = preg_replace($pattern, $replace, $o);
+			$pattern = "/<\s*iframe[^>]*>(.*?)<\s*\/\s*iframe>/i";
+			$o = preg_replace($pattern, $replace, $o);
+		}
+
+		header("Content-type: text/html");
+		echo "<!DOCTYPE html><html><body>\r\n";
+		// We can remove this hack once Internet Explorer recognises HTML5 natively
```

Comment:
The MSIE is dead. Do we really need to have some weird workaround here?

Reply:
It doesn't hurt, so I left it.